cassandra-analytics-core/src/main/java/org/apache/cassandra/spark/bulkwriter/CassandraClusterInfo.java

```diff
@@ -478,21 +490,21 @@ private Map<String, Set<String>> getPendingReplicas(TokenRangeReplicasResponse r
         // Filter writeReplica entries and the value replicaSet to only include those with pending replicas
         return writeReplicasByDC.entrySet()
                                 .stream()
-                                .filter(e -> e.getValue().stream()
+                                .filter(e -> e.getValue().stream() // todo: transformToHostWithoutPort is called twice for entries
                                               .anyMatch(v -> pendingReplicas.contains(transformToHostWithoutPort(v))))
                                 .collect(Collectors.toMap(Map.Entry::getKey,
                                                           e -> e.getValue().stream()
                                                                 .filter(v -> pendingReplicas.contains(transformToHostWithoutPort(v)))
                                                                 .collect(Collectors.toSet())));
     }
 
-    private String transformToHostWithoutPort(String v)
+    private static String transformToHostWithoutPort(String v)
```

Comment:
not ipv6 friendly?

Reply:
talked offline. decide to not touch it as it is unrelated with the patch.